lib/cosmos/interfaces/protocols/template_stream_protocol.rb

```diff
@@ -141,7 +143,7 @@ def read_packet(packet)
 
         return result_packet
       else
-        return :STOP
+        return packet
```

Comment:
Good catch on the shift calls above.   I'm torn on returning packet here rather than :STOP though.   The TEMPLATE protocol does not really support non cmd/response style packets.   I guess this would at least record some UNKNOWNs which could be helpful for debugging, so I guess it is a good change.